addrs.go

```diff
@@ -45,17 +55,29 @@ func ParseWebsocketNetAddr(a net.Addr) (ma.Multiaddr, error) {
 		return nil, fmt.Errorf("not a websocket address")
 	}
 
-	tcpaddr, err := net.ResolveTCPAddr("tcp", wsa.Host)
-	if err != nil {
-		return nil, err
+	// Detect if host is IP address or DNS
+	var tcpma ma.Multiaddr
+	if ip := net.ParseIP(wsa.Hostname()); ip != nil {
```

Comment:
I had to change this code because we cannot lookup DNS addresses in the browser. `ResolveTCPAddr` calls `lookupIP` which is [not implemented in JavaScript/Wasm environments](https://github.com/golang/go/blob/e948d2b73ede67f12bff9e4d050f0e1425163010/src/net/lookup_fake.go#L22). This code seems to work but I still have two questions:

1. Does libp2p _need_ to resolve the DNS name here for some reason?
2. Suppose `a` contains a domain name instead of an IP address. How can I differentiate between dns4 and dns6?

Reply:
> Does libp2p need to resolve the domain name here for some reason?

No, and it shouldn't resolve it.

> Suppose a contains a domain name instead of an IP address. How can I differentiate between dns4 and dns6?

You can now use `/dns`.

Reply:
Oh, resolved below.